camel-core/src/main/java/org/apache/camel/impl/transformer/ProcessorTransformer.java

```diff
@@ -0,0 +1,122 @@
+ */
+public class ProcessorTransformer extends Transformer {
+    private static final Logger LOG = LoggerFactory.getLogger(ProcessorTransformer.class);
+
+    private Processor processor;
+    private String transformerString;
+
+    public ProcessorTransformer(CamelContext context) {
+        setCamelContext(context);
+    }
+
+    /**
+     * Perform data transformation with specified from/to type using Processor.
+     * @param message message to apply transformation
+     * @param from 'from' data type
+     * @param to 'to' data type
+     */
+    @Override
+    public void transform(Message message, DataType from, DataType to) throws Exception {
+        Exchange exchange = message.getExchange();
+        CamelContext context = exchange.getContext();
+        if (from.isJavaType()) {
+            Object input = message.getBody();
+            Class<?> fromClass = context.getClassResolver().resolveClass(from.getName());
+            if (!fromClass.isAssignableFrom(input.getClass())) {
+                LOG.debug("Converting to '{}'", fromClass.getName());
+                input = context.getTypeConverter().mandatoryConvertTo(fromClass, input);
+                message.setBody(input);
+            }
+        }
+        
+        LOG.debug("Sending to transform processor '{}'", processor);
+        DefaultExchange transformExchange = new DefaultExchange(exchange);
+        transformExchange.setIn(message);
+        transformExchange.setProperties(exchange.getProperties());
+        processor.process(transformExchange);
+        Message answer = transformExchange.hasOut() ? transformExchange.getOut() : transformExchange.getIn();
+        
+        if (to.isJavaType()) {
+            Object answerBody = answer.getBody();
+            Class<?> toClass = context.getClassResolver().resolveClass(to.getName());
+            if (!toClass.isAssignableFrom(answerBody.getClass())) {
+                LOG.debug("Converting to '{}'", toClass.getName());
+                answerBody = context.getTypeConverter().mandatoryConvertTo(toClass, answerBody);
+                answer.setBody(answerBody);
+            }
+        }
+        
+        message.copyFrom(answer);
+    }
+
+    /**
+     * Set Processor.
+     * @param processor Processor
+     * @return this ProcessorTransformer instance
+     */
+    public ProcessorTransformer setProcessor(Processor processor) {
+        this.processor = processor;
+        this.transformerString = null;
+        return this;
+    }
+
+    @Override
+    public String toString() {
+        if (transformerString == null) {
+            transformerString =
+                String.format("ProcessorTransformer[scheme='%s', from='%s', to='%s', processor='%s']",
+                    getModel(), getFrom(), getTo(), processor);
+        }
+        return transformerString;
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        if (this.processor instanceof ServiceSupport) {
```

Comment:
Use ServiceHelper.startService / and ditto for stop. No need for instanceof check either